File: frontend/web_ccadmin_ecommerce/src/app/enterprise/client/model/dto/ShippingPriceDto.ts
import { DeliveryCoverageDto } from './DeliveryCoverageDto';
import { ShippingScheduleDto } from './ShippingScheduleDto';

export class ShippingPriceRequestDto {
  public StoreCod: string = '';
  public DeliveryTypeCod: string = '';
  public Latitude: number | null = null;
  public Longitude: number | null = null;
  public CountryCod: string = '';
  public UbigeoCod: string = '';
}

export class ShippingPriceDto {
  public DeliveryTypeCod: string = '';
  public ScheduleType: string = '';
  public ShippingConfigCod: string = '';
  public ProductCod: string = '';
  public Description: string = '';
  public DistanceKm: number = 0;
  public PriceBase: number = 0;
  public PricePerKm: number = 0;
  public Amount: number = 0;
  public Coverage: DeliveryCoverageDto = new DeliveryCoverageDto();
  public Schedule: ShippingScheduleDto | null = null;
}

File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/delivery/model/dto/ShippingPriceDto.java
package com.ccadmin.app.delivery.model.dto;

import java.math.BigDecimal;

public class ShippingPriceDto {

    public String DeliveryTypeCod;
    public String ScheduleType;
    public String ShippingConfigCod;
    public String ProductCod;
    public String Description;
    public BigDecimal DistanceKm;
    public BigDecimal PriceBase;
    public BigDecimal PricePerKm;
    public BigDecimal Amount;
    public DeliveryCoverageDto Coverage;
    public ShippingScheduleDto Schedule;
}
